new-site/docs-site/content/0.19.0/guide/typesense-dynamodb.md

```diff
@@ -0,0 +1,240 @@
+}
+```
+
+The policy has three statements that allow TypesenseLambdaRole to do the following:
+
+* Run a Lambda function ```typesense-indexing```. We'll be creating the function later in this tutorial.
+* Access Amazon CloudWatch Logs. The Lambda function writes diagnostics to CloudWatch Logs at runtime.
+* Read data from the DynamoDB stream for ```typesense```.
+
+Now, we are going to attach the above roles to our IAM execution role which we have created
+
+```bash
+aws iam put-role-policy --role-name TypesenseLambdaRole \
+    --policy-name TypesenseLambdaRolePolicy \
+    --policy-document file://role-policy.json
+```
+## Step 4: Create a Lambda Function
+
+Head over to Lambda section of AWS and create a new Lambda function with the above created execution role.
+
+Now, let's create a new python script file named `lambda_function.py`. For creating collections and indexing into your cloud cluster refer: [Typesense API](../api/README.md)
+
+```python
+def lambda_handler(event, context):
+    # Your code goes here
+    return response
+```
+
+A example code snippet for the function,
+
+```python
+client = typesense.Client({
+    'nodes': [{
+        'host': '<Endpoint URL>',
+        'port': '<Port Number>',
+        'protocol': 'https',
+    }],
+    'api_key': '<API Key>',
+    'connection_timeout_seconds': 2
+})
+
+processed = 0
+for record in event['Records']:
+    ddb_record = record['dynamodb']
+    if record['eventName'] == 'REMOVE':
+        res = client.collections['<collection-name>'].documents[str(ddb_record['OldImage']['id']['N'])].delete()
+    else:
+        upload = ddb_record['NewImage'] # format your document here and the use upsert function to index it.
+        res = client.collections['<collection-name>'].upsert(upload)
+        print(res)
+    processed = processed + 1
+
+print('Successfully processed {} records'.format(processed))
+```
+
+Note: Install all your code using `pip install <dependency-name> -t .`. This will install all the dependencies for the function in the current directory.
+
+After this zip your current directory and upload it to your Lambda function.
+
+You can also do this using AWS CLI,
+
+* Get the ARN for the the execution role you created.
+    ```bash
+    aws iam get-role --role-name TypesenseLambdaRole
+    ```
+    In output, look for ARN,
+    ```json
+    ...
+    "Arn": "arn:aws:iam::region:role/service-role/TypesenseLambdaRole"
+    ...
+    ```
+
+* Now, create the Lambda function
+
+    ```bash
+    aws lambda create-function \
+    --region us-east-2 \
+    --function-name typesense-indexing \
+    --zip-file fileb://typesense-indexing.zip \
+    --role `roleARN` \
+    --handler lambda_function.lambda_handler \
+    --timeout 5 \
+    --runtime python3.7
+    ```
+
+Make sure to test the function with a set of sample events before enabling the trigger.
+
+## Step 5: Setup up a trigger
+
+Now, navigate to DynamoDB table &#8594; Triggers and add this existing Lambda function to that table.
```

Comment:
Now, navigate to your DynamoDB table in the AWS Console, then visit the Triggers section...